assets/app/scripts/directives/fromFile.js

```diff
@@ -236,11 +236,14 @@ angular.module("openshiftConsole")
           return DataService.get(resourceGroupVersion, item.metadata.name, $scope.context, {errorNotification: false}).then(
             // resource does exist
             function(resource) {
-              if (item.kind !== "Template") {
-                resource.spec = item.spec;
-              }
-              resource.metadata.annotations = item.metadata.annotations;
-              resource.metadata.labels = item.metadata.labels;
+              _.each(item, function(value, key) {
+                if (key === "metadata") {
+                  resource.metadata.annotations = item.metadata.annotations;
+                  resource.metadata.labels = item.metadata.labels;
+                } else {
+                  resource[key] = value;
+                }
+              })
```

Comment:
If there are properties on the old resource, but not the new, they won't be deleted. That seems like a problem.

@fabianofranz can tell us what the algorithm is for `oc replace`. I want to make sure behavior is consistent.